### ISSUE TITLE
Firefox Private Browsing: A mutation operation was attempted on a database that did not allow mutations

### ISSUE THREAD

**Issue by boldtrn** [firebase/firebase-js-sdk#4069]
* Operating System version: Ubuntu
  * Browser version: Firefox 82.0.3 (Private Browsing)
  * Firebase SDK version: 8.0.2
  * Firebase Product: auth, firestore (auth, database, storage, etc)

#### Steps to reproduce:

When opening the website with Firefox private browsing the following error is thrown:

`Uncaught Error: InvalidStateError: A mutation operation was attempted on a database that did not allow mutations.`

This is thrown here:
```
function uk(a) {
    return new D(function (b, c) {
      var d = a.v.open("firebaseLocalStorageDb", 1);

      d.onerror = function (e) {
        try {
          e.preventDefault();
        } catch (f) {}

        c(Error(e.target.error));
      };
```

**Comment by hsubox76** [comment]
This seems to be in auth code accessing indexedDB.

**Comment by jdpowell1** [comment]
I'm getting the same issue in Firebox 82.0.3 Private Browsing with SDK version 8.0.2 on Windows 10.

**Comment by ckybonist** [comment]
> Triage note: This seems to be in auth code accessing indexedDB.

Is the error raised from here?
https://github.com/firebase/firebase-js-sdk/blob/master/packages/auth/src/storage/indexeddb.js#L298

**Comment by jahed** [comment]
Worth mentioning that this seems to be a fatal error. There doesn't seem to be a way to catch errors when calling `firebase.auth()` to fail authentication. Auth doesn't have any error hooks so anything waiting on it will hang indefinitely.

I guess the only workaround is to wrap any firebase.auth calls in a timeout since there's a chance it'll hit a silent dead end.

**Comment by aliokan** [comment]
Is someone find a solution to `firebase.getAuth()` signIn on Firefox Private window?

**Comment by kechpaja** [comment]
@aliokan in my own project at least, this bug only appears in very recent versions. If you step back through the availalbe versions until the bug goes away, you should be able to mitigate it for the the time being.

**Comment by harshit-j** [comment]
This bug appeared in 8.0.2 release https://firebase.google.com/support/release-notes/js#version_802_-_november_12_2020
Probably because of this fix: https://github.com/firebase/firebase-js-sdk/issues/1926

**Comment by hsubox76** [comment]
I think this may be the same bug as https://github.com/firebase/firebase-js-sdk/issues/4113. I believe Firefox's message for that same error ("IDBFactory.open() called in an invalid security context") is the text reported in this issue ("A mutation operation was attempted on a database that did not allow mutations.")

**Comment by samhorlbeck** [comment]
Hi everyone, thanks for your patience. This should be fixed by https://github.com/firebase/firebase-js-sdk/pull/4146, which is included in `8.1.2`. Please upgrade to `8.1.2` for the fix. Thanks!
